scripts/build.ps1

```diff
@@ -69,7 +69,7 @@ param (
     [string]$Arch = "",
 
     [Parameter(Mandatory = $false)]
-    [ValidateSet("gamecore_console", "uwp", "windows", "linux", "macos", "android", "ios", "winkernel")] # For future expansion
+    [ValidateSet("gamecore_console", "uwp", "windows", "linux", "macos", "android", "ios", "winkernel", "windows-vs")] # For future expansion
```

Comment:
Exposing this new build option as a new "Platform" might be confusing. windows user mode built with "windows" platform will be equivalent to windows user mode built with "windows-vs" platform. They will both be usable for win32 apps.

I'd suggest introducing a "BuildToolSet" or similar that allows you to specify a specific build tool set within a platform.

.\build.ps1 -Platform windows -BuildToolSet vs
or 
.\build.ps1 -Platform windows -BuildToolSet cmake

Reply:
good idea

Reply:
There is already a Clang switch that is related to this. Hmm, I'm not exactly sure what the best way is to expose all of these related options. I think we can do better than extending "Platform" though

Reply:
Added BuildToolSet